src/connmgr.cpp

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2009-2010 Satoshi Nakamoto
+// Copyright (c) 2009-2015 The Bitcoin Core developers
+// Copyright (c) 2016-2017 The Bitcoin Unlimited developers
+
+#include <atomic>
+
+#include "connmgr.h"
+
+
+NodeId CConnMgr::nextNodeId()
+{
+    static std::atomic<NodeId> next;
+
+    // Pre-increment; do not use zero
+    return ++next;
+}
+
+std::unique_ptr<CConnMgr> connmgr(new CConnMgr);
```

Comment:
Where is this guy being used? Or is there something planned? You seem to only use static methods otherwise.